coalahtml/tree.py

```diff
@@ -0,0 +1,69 @@
+from collections import OrderedDict
+from queue import Queue
+import os
+
+
+class Node:
+
+    def __init__(self, name):
+        self.child = {}
+        self.parent = None
+        self.name = name
+        self.abspath = ""
+
+
+class Tree:
+
+    def __init__(self, file_list):
+        """
+        Create a root and build the graph.
+        """
+        self.root = Node(Tree.findroot(file_list))
+        self.root.abspath = self.root.name
+        self.fl_split = [path.split(os.sep) for path in file_list]
+        self._build(self.fl_split)
+
+    @staticmethod
+    def findroot(file_list):
+        prefix = os.path.commonprefix(file_list)
+        last_sep = prefix.rfind(os.sep)
+        return prefix[:last_sep]
+
+    def _add_node(self, new_node, prev_node, head):
+        if prev_node.child.get(new_node.name, None) is None:
+            prev_node.child[new_node.name] = new_node
+        else:
+            # Duplicates
+            return prev_node.child[new_node.name]
+
+        return new_node
```

Comment:
This can be rewritten as follows:
```python
if not prev_node.child.get(new_node.name):
    prev_node.child[new_node.name] = new_node
return prev_node.child[new_node.name]
```
Also this function is very simple, so I'm not it is needed. Basically it's one statement inside the `if` and you use this only in one place.